scene/resources/texture.cpp

```diff
@@ -1867,6 +1868,266 @@ Ref<Image> GradientTexture::get_image() const {
 	return RenderingServer::get_singleton()->texture_2d_get(texture);
 }
 
+GradientTexture2D::GradientTexture2D() {
+	_queue_update();
+}
+
+GradientTexture2D::~GradientTexture2D() {
+	if (texture.is_valid()) {
+		RS::get_singleton()->free(texture);
+	}
+}
+
+void GradientTexture2D::set_gradient(Ref<Gradient> p_gradient) {
+	if (gradient == p_gradient) {
+		return;
+	}
+	if (gradient.is_valid()) {
+		gradient->disconnect(CoreStringNames::get_singleton()->changed, callable_mp(this, &GradientTexture2D::_queue_update));
+	}
+	gradient = p_gradient;
+	if (gradient.is_valid()) {
+		gradient->connect(CoreStringNames::get_singleton()->changed, callable_mp(this, &GradientTexture2D::_queue_update));
+	}
+	_queue_update();
+}
+
+Ref<Gradient> GradientTexture2D::get_gradient() const {
+	return gradient;
+}
+
+void GradientTexture2D::_queue_update() {
+	if (update_pending) {
+		return;
+	}
+	update_pending = true;
+	call_deferred("_update");
+}
+
+void GradientTexture2D::_update() {
+	update_pending = false;
+
+	if (gradient.is_null()) {
+		return;
+	}
+	Ref<Image> image;
+	image.instantiate();
+
+	if (gradient->get_points_count() <= 1) { // No need to interpolate.
+		image->create(width, height, false, (use_hdr) ? Image::FORMAT_RGBAF : Image::FORMAT_RGBA8);
+		Color color;
+		if (gradient->get_points_count() == 1) {
+			color = gradient->get_color(0);
+		} else {
+			color = Color(0, 0, 0, 1);
+		}
+		image->fill(color);
```

Comment:
```suggestion
		image->create(width, height, false, use_hdr ? Image::FORMAT_RGBAF : Image::FORMAT_RGBA8);
		if (gradient->get_points_count() == 1) {
			image->fill(gradient->get_color(0));
		}
```
Or if black is preferred over transparent:
```suggestion
		image->create(width, height, false, use_hdr ? Image::FORMAT_RGBAF : Image::FORMAT_RGBA8);
		image->fill(gradient->get_points_count() == 1 ? gradient->get_color(0) : Color(0, 0, 0, 1));
```
Just a suggestion.